test/blockgaslimit.js

```diff
@@ -0,0 +1,119 @@
+const Web3 = require("web3");
+const assert = require("assert");
+const Ganache = require("../index");
+const path = require("path");
+const compileAndDeploy = require("./helpers/contracts").compileAndDeploy;
+
+const TARGET = process.env.TARGET || "ganache";
+const PARITY_UNLOCKED_ADDR = process.env.PARITY_UNLOCKED_ADDR || "0x00a329c0648769A73afAc7F9381E08FB43dBEA72";
+const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+
+function setUp(options = { mnemonic }, contractName = "BlockGasLimit") {
+  const context = {
+    options: options,
+    provider: null,
+    web3: null,
+    accounts: [],
+    contractArtifact: {},
+    instance: null
+  };
+
+  before("setup web3", async function() {
+    // Enable Ganache provider
+    if (TARGET === "ganache") {
+      // eslint-disable-next-line new-cap
+      context.provider = new Ganache.provider(context.options);
+      context.options.blockTime = 2000;
+      context.web3 = new Web3(context.provider);
+    }
+
+    // Enable Geth provider
+    if (TARGET === "geth") {
```

Comment:
There is a lot of code duplication between the multiple test targets here so I think these should all be refactored so there isn't so much duplication between them.

Also, we'll soon be migrating nearly all of our tests to support multiple targets (like you've done here) and this multi target configuration you've made would be very useful to have available in a reusable `./helpers/` file. The tricky part will be that we'll need some sort of automated mechanism for spooling up geth and parity locally for testing.

Could you pull your `before("setup web3, fn)" functionality out into it's own file that would could reuse?

Reply:
I can start placing more focus on a refactoring effort. Also decided to use code duplication to help validate my local Geth and Parity environments. Figured it would be a good setup for other until we all share a consistent geth/parity environment.